fabric-client-gametest-api-v1/src/client/java/net/fabricmc/fabric/impl/client/gametest/ClientGameTestContextImpl.java

```diff
@@ -0,0 +1,314 @@
+
+		for (int i = 0; i < ticks; i++) {
+			ThreadingImpl.runTick();
+		}
+	}
+
+	@Override
+	public void waitFor(Predicate<MinecraftClient> predicate) {
+		ThreadingImpl.checkOnGametestThread("waitFor");
+		Preconditions.checkNotNull(predicate, "predicate");
+		waitFor(predicate, DEFAULT_TIMEOUT);
+	}
+
+	@Override
+	public void waitFor(Predicate<MinecraftClient> predicate, int timeout) {
+		ThreadingImpl.checkOnGametestThread("waitFor");
+		Preconditions.checkNotNull(predicate, "predicate");
+
+		if (timeout == NO_TIMEOUT) {
+			while (!computeOnClient(predicate::test)) {
+				ThreadingImpl.runTick();
+			}
+		} else {
+			Preconditions.checkArgument(timeout > 0, "timeout must be positive");
+
+			for (int i = 0; i < timeout; i++) {
+				if (computeOnClient(predicate::test)) {
+					return;
+				}
+
+				ThreadingImpl.runTick();
+			}
+
+			if (!computeOnClient(predicate::test)) {
+				throw new AssertionError("Timed out waiting for predicate");
+			}
+		}
+	}
+
+	@Override
+	public void waitForScreen(@Nullable Class<? extends Screen> screenClass) {
+		ThreadingImpl.checkOnGametestThread("waitForScreen");
+
+		if (screenClass == null) {
+			waitFor(client -> client.currentScreen == null);
+		} else {
+			waitFor(client -> screenClass.isInstance(client.currentScreen));
+		}
+	}
+
+	@Override
+	public void setScreen(@Nullable Screen screen) {
+		ThreadingImpl.checkOnGametestThread("setScreen");
+		runOnClient(client -> client.setScreen(screen));
+	}
+
+	@Override
+	public void clickScreenButton(String translationKey) {
+		ThreadingImpl.checkOnGametestThread("clickScreenButton");
+		Preconditions.checkNotNull(translationKey, "translationKey");
+
+		runOnClient(client -> {
+			if (!tryClickScreenButtonImpl(client.currentScreen, translationKey)) {
+				throw new AssertionError("Could not find button '%s' in screen '%s'".formatted(
+					translationKey,
+					Nullables.map(client.currentScreen, screen -> screen.getClass().getName())
+				));
+			}
+		});
+	}
+
+	@Override
+	public boolean tryClickScreenButton(String translationKey) {
+		ThreadingImpl.checkOnGametestThread("tryClickScreenButton");
+		Preconditions.checkNotNull(translationKey, "translationKey");
+
+		return computeOnClient(client -> tryClickScreenButtonImpl(client.currentScreen, translationKey));
+	}
+
+	private static boolean tryClickScreenButtonImpl(@Nullable Screen screen, String translationKey) {
+		if (screen == null) {
+			return false;
+		}
+
+		final String buttonText = Text.translatable(translationKey).getString();
+		final ScreenAccessor screenAccessor = (ScreenAccessor) screen;
+
+		for (Drawable drawable : screenAccessor.getDrawables()) {
+			if (drawable instanceof PressableWidget pressableWidget && pressMatchingButton(pressableWidget, buttonText)) {
+				return true;
+			}
+
+			if (drawable instanceof Widget widget) {
+				widget.forEachChild(clickableWidget -> pressMatchingButton(clickableWidget, buttonText));
```

Comment:
Not a new issue, but I thought I'd mention it here: If the target button only exists as a child of a non-pressable `Widget`, then `tryClickScreenButtonImpl()` will press that button and then return `false` as if it didn't find it. The return value of `pressMatchingButton()` goes nowhere in that case.

I can't find an example where this would actually happen though. Minecraft usually calls `widget.forEachChild(this::addDrawableChild);`, so any button that would be affected by this bug is found directly in the `drawables` list instead.